### PR TITLE
fix: matching algorithm

### PR DESCRIPTION
**Pull request by tituschewxj**
Hotfix the matched difficulties and topics so that it will be an OR if either is empty. Also fixes an issue with difficulties matching using wrong variable topics.

### CODE REVIEW

File: apps/matching-service/processes/findmatches.go [hunk 1]
```diff
@@ -74,18 +74,38 @@ func foundMatch(tx *redis.Tx, ctx context.Context, currentUser *models.MatchRequ
 	}
 
 	var matchedTopics []string
-	for _, topic := range currentUser.Topics {
+	if len(currentUser.Topics) == 0 || len(currentUser.Topics) == 0 {
```
Comment (bensohh) [inline apps/matching-service/processes/findmatches.go]: Can double-check why the if check contains two of the same clause?

File: apps/matching-service/processes/findmatches.go [hunk 2]
```diff
@@ -74,18 +74,38 @@ func foundMatch(tx *redis.Tx, ctx context.Context, currentUser *models.MatchRequ
...
+				}
 			}
 		}
 	}
 	var matchedDifficulties []string
-	for _, topic := range currentUser.Difficulties {
-		for _, otherTopic := range matchedUser.Difficulties {
-			if topic == otherTopic {
-				matchedDifficulties = append(matchedDifficulties, topic)
+	if len(currentUser.Difficulties) == 0 || len(currentUser.Difficulties) == 0 {
```
Comment (bensohh) [inline apps/matching-service/processes/findmatches.go]: same for this

**Review by bensohh — APPROVED** [review]
If required, can make the necessary duplicate changes before merging thanks @tituschewxj

File: apps/matching-service/processes/findmatches.go (hunk 1 above)
Comment (tituschewxj) [reply apps/matching-service/processes/findmatches.go]: Good catch, it should be `matchedUser.Topics` instead.

**Pull request merged**